piecrust/src/call_tree.rs

```diff
@@ -209,8 +375,47 @@ impl CallTree {
         }
     }
 
-    /// Returns an iterator over the call tree, starting from the rightmost
-    /// leaf, and proceeding to the top of the current position of the tree.
+    /// Returns an iterator over the current node and its descendants.
+    ///
+    /// Traverses in **reverse post-order** (rightmost leaf first):
+    /// deepest-rightmost children first, then left siblings, then parents.
+    /// This matches how contract calls unwind (deepest calls complete first).
+    ///
+    /// # Example
+    ///
+    /// For tree `A -> B -> [D, E]` and `A -> C`, if positioned at A:
+    /// A calls B, which calls D then E, then A calls C.
+    ///
+    /// Iterator yields: C, E, D, B, A (rightmost leaves first)
+    ///
+    /// **Notation**:
+    /// - `->` means "calls"
+    /// - `[]` groups sibling calls (sequential calls made by the same contract)
+    /// - `,` separates siblings (calls made one after another)
```

Comment:
I'd move the notations up and add the structure like below:
```suggestion
    /// **Notation**:
    /// - `->` means "calls"
    /// - `[]` groups sibling calls (sequential calls made by the same contract)
    /// - `,` separates siblings (calls made one after another)
    ///
    /// For tree `A -> B -> [D, E]` and `A -> C`, if positioned at A:
    /// A calls B, which calls D then E, then A calls C.
    ///
    /// Tree structure:
    /// ```text
    /// ...
    /// ```
    ///
    /// Iterator yields: C, E, D, B, A (rightmost leaves first)
```